biz.aQute.bndlib/src/aQute/bnd/cdi/CDIAnnotations.java

```diff
@@ -83,6 +84,19 @@ public boolean analyzeJar(Analyzer analyzer) throws Exception {
 			}
 		}
 
+		// If, after removing every instruction where 'discover=none', there are
+		// no items left we should just return.
+		if (header.entrySet()
+			.stream()
+			.map(e -> e.getValue()
+				.get("discover"))
+			.filter(Objects::nonNull)
+			.filter(discover -> Discover.valueOf(discover) == Discover.none)
+			.count() == 0) {
```

Comment:
Instead of `count() == 0`, just replace the final `filter` call with:

```java
.allMatch(discover -> Discover.valueOf(discover) == Discover.none);
```
But handling the IllegalArgumentException issue.

`allMatch` will early complete with `false` if a non-`none` value is found.